iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/client/HiveRESTCatalogClient.java

```diff
@@ -228,4 +231,9 @@ public void dropDatabase(DropDatabaseRequest req) {
     validateCurrentCatalog(req.getCatalogName());
     restCatalog.dropNamespace(Namespace.of(req.getName()));
   }
+
+  @Override
+  public LockResponse lock(LockRequest request) {
+    return new LockResponse(request.getTxnid(), LockState.ACQUIRED);
+  }
```

Comment:
What would happen if we implement this? My local test succeeded without this change